test/SILGen/apply_abstraction_nested.swift

```diff
@@ -1,11 +1,11 @@
 
-// RUN: %target-swift-frontend -enable-sil-ownership -emit-silgen %s -swift-version 3 | %FileCheck %s
+// RUN: %target-swift-frontend -enable-sil-ownership -emit-silgen %s | %FileCheck %s
 
-infix operator ~> { precedence 255 associativity left }
+infix operator ~>
 
 protocol P { }
 
-func bar<T:P>(_: inout T) -> () -> () { return {_ in ()} }
+func bar<T:P>(_: inout T) -> (()) -> () { return {_ in ()} }
```

Comment:
This looks like a different test to me.

Reply:
The result of bar() is passed to a generic function that takes (T) -> U. It's effectively the same because SIL recursively destructures tuples